components/database/mariadb-103/Makefile

```diff
@@ -0,0 +1,138 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2011-2013, EveryCity Ltd. All rights reserved.
+# Copyright 2016-2017, Wiselabs Software Ltda - http://www.wiselabs.com.br . All rights reserved.
+# Copyright 2018, Michal Nowak
+#
+
+PREFERRED_BITS=64
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		mariadb
+COMPONENT_MAJOR_VERSION=10.3
+COMPONENT_MINOR_VERSION=8
+COMPONENT_VERSION=	$(COMPONENT_MAJOR_VERSION).$(COMPONENT_MINOR_VERSION)
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_PROJECT_URL=	https://www.mariadb.org/
+COMPONENT_ARCHIVE_HASH=	sha256:30bec17514d12d811a05757be46bd41724df28002cdec550d5c757b0b3f31ab8
+COMPONENT_ARCHIVE_URL=	https://downloads.mariadb.com/MariaDB/$(COMPONENT_SRC)/source/$(COMPONENT_ARCHIVE)
+COMPONENT_SIG_URL=	$(COMPONENT_ARCHIVE_URL).asc
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/cmake.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+CPPFLAGS += -I/usr/include/pcre
+
+CONFIGURE_PREFIX=/usr/mariadb/$(COMPONENT_MAJOR_VERSION)
+CONFDIR=/etc/mariadb/$(COMPONENT_MAJOR_VERSION)
+DATA_PREFIX=/var/mariadb/$(COMPONENT_MAJOR_VERSION)
+
+CMAKE_PLUGINDIR.64 =	lib/$(MACH64)/plugin
+CMAKE_PLUGINDIR.32 =	lib/$(MACH32)/plugin
+
+CMAKE_OPTIONS+= -DCMAKE_C_COMPILER=$(CC)
+CMAKE_OPTIONS+= -DCMAKE_CXX_COMPILER=$(CXX)
+CMAKE_OPTIONS+= -DCMAKE_C_FLAGS=$(CFLAGS)
+# `-fpermissive` due to:
+# 	sql/sql_class.h:6416:42: warning: invalid conversion from 'const char*' to 'char*' [-fpermissive]
```

Comment:
Should we add patch to use explicit conversion?

Reply:
Workaround removed, for changes see `patches/08-dtrace-invalid-conversion.patch`.